src/Type/Php/MinMaxFunctionReturnTypeExtension.php

```diff
@@ -128,6 +135,50 @@ private function processType(
 		return $resultType;
 	}
 
+	private function mergeRangeTypes(
+		Type $firstType,
+		Type $secondType,
+		string $functionName
+	): ?Type
+	{
+		$rangeType = $intType = null;
+
+		if (
+			$firstType instanceof IntegerRangeType
+			&& $secondType instanceof ConstantIntegerType
+		) {
+			$rangeType = $firstType;
+			$intType = $secondType;
+		}
```

Comment:
I'd rather have this implemented with `isSuperTypeOf` logic instead of doing `instanceof *Type`. You're cover more ground with that. Learn more: https://phpstan.org/developing-extensions/type-system

Reply:
I tried several variants, but didn't find one with the same coverage this currently implements.
`isSuperTypeOf` does not work on `IntegerRangeType` as I cannot instantiate a IntegerRangeType.

the logic I use here only works for `ConstantIntegerType` in combination with `IntegerRangeType`.
I have no idea how I can express this properly with `isSuperTypeOf` for the code at hand, without additonal `instanceof *Type` checks